libs/ngxtension/computed-async/src/computed-async.ts

```diff
@@ -122,31 +257,48 @@ export function computedAsync<T>(
 			options?.behavior ?? 'switch',
 		);
 
-		const sourceResult = source$.subscribe({
-			next: (value) => sourceValue.set(value),
-			error: (error) => {
+		const sourceResult = source$
+			// we skip the first value if requireSync is true because we already set the value in the sourceValue
+			.pipe(options?.requireSync ? skip(1) : (x) => x)
+			.subscribe({
+				next: (value) => sourceValue.set({ kind: StateKind.Value, value }),
 				// NOTE: Error should be handled by the user (using catchError or .catch())
-				sourceValue.set(error);
-			},
-		});
+				error: (error) => sourceValue.set({ kind: StateKind.Error, error }),
+			});
 
 		destroyRef.onDestroy(() => {
 			effectRef.destroy();
 			sourceResult.unsubscribe();
 		});
 
+		if (options?.requireSync && sourceValue().kind === StateKind.NoValue) {
+			throw new Error(REQUIRE_SYNC_ERROR_MESSAGE);
+		}
+
 		// we return a computed value that will return the current value
 		// in order to support the same API as computed()
 		return computed(
 			() => {
-				const value: T = sourceValue()!;
-				return value;
+				const state = sourceValue();
+				switch (state.kind) {
+					case StateKind.Value:
+						return state.value;
+					case StateKind.Error:
+						throw state.error;
+					case StateKind.NoValue:
+						throw new Error(REQUIRE_SYNC_ERROR_MESSAGE);
```

Comment:
question: why do we throw `REQUIRE_SYNC` error here without checking for `options.requiredSync`?

Reply:
Can you add some comments for that case then?